warehouse/query-core/src/main/java/datawave/query/config/ShardQueryConfiguration.java

```diff
@@ -230,152 +248,318 @@
-    
     private boolean indexOnlyFilterFunctionsEnabled = false;
-    
     private boolean compositeFilterFunctionsEnabled = false;
     
-    protected int maxScannerBatchSize = 2000;
-    
-    /**
-     * Index batch size is the size of results use for each index lookup
-     */
-    protected int maxIndexBatchSize = 200;
-    
-    protected boolean allTermsIndexOnly;
-    
-    protected String password;
-    
-    protected long maxIndexScanTimeMillis = Long.MAX_VALUE;
-    
-    protected boolean collapseUids = false;
-    
-    protected boolean sequentialScheduler = false;
-    
-    protected boolean collectTimingDetails = false;
-    
-    protected boolean logTimingDetails = false;
-    
-    protected boolean sendTimingToStatsd = true;
-    protected String statsdHost = "localhost";
-    protected int statsdPort = 8125;
-    protected int statsdMaxQueueSize = 500;
-    
-    protected boolean limitAnyFieldLookups = true;
-    
-    protected Set<String> groupFields;
-    
     protected int groupFieldsBatchSize;
-    
-    protected Set<String> uniqueFields;
-    
     private boolean accrueStats = false;
-    
+    private Set<String> groupFields = new HashSet<>(0);
+    private Set<String> uniqueFields = new HashSet<>(0);
     private boolean cacheModel = false;
-    
-    protected boolean bypassExecutabilityCheck = false;
-    
     /**
-     * Allows for back off of scanners.
-     */
-    protected boolean backoffEnabled = false;
-    
-    /**
-     * Allows for the unsorted UIDs feature (see SortedUIDsRequiredVisitor)
-     */
-    protected boolean unsortedUIDsEnabled = true;
-    
-    /**
-     * Allows us to serialize query iterator
-     */
-    protected boolean serializeQueryIterator = false;
-    
-    /**
-     * Used to enable the SourceThreadTrackingIterator on the tservers
-     */
-    protected boolean debugMultithreadedSources = false;
-    
-    /**
-     * Used to enable Event Field Value filtering in the TLD based on Query Expressions
+     * should the sizes of documents be tracked for this query
      */
-    protected boolean dataQueryExpressionFilterEnabled = false;
+    private boolean trackSizes = true;
     
-    /**
-     * Used to enable sorting query ranges from most to least granular for queries which contain geowave fields in ThreadedRangeBundler
-     */
-    protected boolean sortGeoWaveQueryRanges = false;
+    public ShardQueryConfiguration() {
+        super();
+        query = new QueryImpl();
+    }
     
     /**
-     * Used to determine how many ranges the ThreadedRangeBundler should buffer before returning a range to the caller
+     * Simple copy constructor, persists all variables in the ShardQueryConfiguration. Will honor overrides provided via supplied query parameters.
+     *
+     * @param other
+     *            - another ShardQueryConfiguration
      */
-    protected int numRangesToBuffer = 0;
+    public ShardQueryConfiguration(ShardQueryConfiguration other) {
+        
```

Comment:
Is there some way to create a copy that guarentees that everything is actually copied?  Perhaps using BeanUtils or something like that?  Performance should not be as much of an issue.

Reply:
..added it in.